### DISCUSSION TITLE
Connection Name not available when deprecated feature management_metrics_collection is disabled

### DISCUSSION THREAD

**Discussion by EmperorRXF** [rabbitmq/rabbitmq-server#12833] [Questions] [answered]
### Community Support Policy

- [X] I have read [RabbitMQ's Community Support Policy](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md)
- [X] I agree to provide all relevant information (versions, logs, rabbitmq-diagnostics output, detailed reproduction steps)

### RabbitMQ version used

4.0.3

### Erlang version used

26.2.x

### Operating system (distribution) used

Ubuntu 24.04.1 LTS

### How is RabbitMQ deployed?

Debian package

### rabbitmq-diagnostics status output

See https://www.rabbitmq.com/docs/cli to learn how to use rabbitmq-diagnostics
<details>

```
# PASTE OUTPUT HERE, BETWEEN BACKTICKS
```
</details>

### Logs from node 1 (with sensitive values edited out)

See https://www.rabbitmq.com/docs/logging to learn how to collect logs
<details>

```
# PASTE LOG HERE, BETWEEN BACKTICKS
```
</details>

### Logs from node 2 (if applicable, with sensitive values edited out)

See https://www.rabbitmq.com/docs/logging to learn how to collect logs
<details>

```
# PASTE LOG HERE, BETWEEN BACKTICKS
```
</details>

### Logs from node 3 (if applicable, with sensitive values edited out)

See https://www.rabbitmq.com/docs/logging to learn how to collect logs
<details>

```
# PASTE LOG HERE, BETWEEN BACKTICKS
```
</details>

### rabbitmq.conf

See https://www.rabbitmq.com/docs/configure#config-location to learn how to find rabbitmq.conf file location
<details>

```
# PASTE rabbitmq.conf HERE, BETWEEN BACKTICKS
```
</details>

### Steps to deploy RabbitMQ cluster

- Containerized RabbitMQ with docker

### Steps to reproduce the behavior in question

- Deploy RMQ v4.0.3 with the `management_metrics_collection` feature DISABLED
- Run atleast one consumer to ensure there will be atleast one connection made to the RMQ server
  - Ensure the consumer will specify a Connection Name prop (i.e. as https://www.rabbitmq.com/docs/connections#client-provided-names)
- Visit the Management UI, and then the "Connections" tab, and check if the "Connection Name" appears alongside the connection

### advanced.config

See https://www.rabbitmq.com/docs/configure#config-location to learn how to find advanced.config file location
<details>

```
# PASTE advanced.config HERE, BETWEEN BACKTICKS
```
</details>

### Application code

<details>

```python
# PASTE CODE HERE, BETWEEN BACKTICKS
```
</details>

### Kubernetes deployment file

<details>

```yaml
# Relevant parts of K8S deployment that demonstrate how RabbitMQ is deployed
# PASTE YAML HERE, BETWEEN BACKTICKS
```
</details>

### What problem are you trying to solve?

- When the `management_metrics_collection` feature is ENABLED, I'm able to see the connection_name information
![image](https://github.com/user-attachments/assets/3b757ce2-8b99-42a9-883e-9d23d9357ede)
![image](https://github.com/user-attachments/assets/3c33516a-52ff-440b-bcd6-0ba2b30870d6)

- When the `management_metrics_collection` feature is DISABLED, this information is missing
![image](https://github.com/user-attachments/assets/33d4b232-66bf-4cfb-8249-358f85a4456b)
![image](https://github.com/user-attachments/assets/9521135f-c462-43fe-80e8-c1caaa46bd99)

- I do understand this feature deprecated since a long time now - https://www.rabbitmq.com/release-information/deprecated-features-list
- However, I also think the `connection_name` is an important piece if information to display along the connections in the Management UI
- So is there an alternative way how I can access the `connection_name` while not relying on this deprecated features?

**Comment by michaelklishin** [comment, marked as answer]
User-provided connection name is [emitted together with other connection metrics](https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbit/src/rabbit_reader.erl#L1792-L1798) to be displayed in the management UI.

If you disable metrics in the management plugin, a whole bunch of data will not be available. The ability to disable metrics in that plugin was added with a specific set of users in mind: those who will use a Prometheus format-based monitoring, standard CLI tools, and other options for monitoring.

`rabbitmqctl list_connections` and the Prometheus plugin use completely different code paths, and will display user-provided connection names regardless of the management plugin configuration.

So, you get what you asked for: a very minimalistic set of data about connections, queues, streams, and so on.

**Reply by razvanphp** [reply]
@michaelklishin you mentioned in https://github.com/rabbitmq/rabbitmq-server/discussions/10905#discussioncomment-10649434  that `management_metrics_collection` was un-deprecated? Is this decided?

We also use the `client_properties` for very useful debug data, not sure if those are exported to prometheus metrics by default.
